frontend/src/core/api/entity.js

```diff
@@ -131,7 +131,10 @@ export default class EntityAPI extends APIBase {
     async getOtherLocales(
         entity: number,
         locale: string,
-    ): Promise<Array<OtherLocaleTranslation>> {
+    ): Promise<{
```

Comment:
Since we're repeating this patter a lot, I suggest you change this to `Promise<OtherLocaleTranslations>` and define a new `OtherLocaleTranslations` type:
```js
export type OtherLocaleTranslations = {|
    +preferred: Array<OtherLocaleTranslation>,
    +other: Array<OtherLocaleTranslation>,
|};
```

Reply:
This change brings back the Flow error that the `object literal is incompatible with exact OtherLocaleTranslations`.

Also, would this be the correct typing for the new format of the `translation`: 
```javascript
export type OtherLocaleTranslation = {|
    +locale: {|
        +code: string,
        +locale: string,
        +locale_pk: number,
        +direction: string,
        +script: string,
    |},
    +translation: string,
|};
```
Or would I just set `locale: Object`

Reply:
Make sure you prepend the `OtherLocaleTranslations` with `?` to make it [optional](https://flow.org/en/docs/types/maybe/) when you use it (e.g. `+translations: ?OtherLocaleTranslations`) and then use `null` instead of `{}`.

The `OtherLocaleTranslation` type looks good (using `Object` is cheating hehe), but you'll need to use the same keys as in https://github.com/mozilla/pontoon/pull/1508#discussion_r354178848.